src/Symfony/Component/DependencyInjection/Tests/Compiler/DecoratorServicePassTest.php

```diff
@@ -115,6 +115,37 @@ public function testProcessWithPriority()
         $this->assertNull($quxDefinition->getDecoratedService());
     }
 
+    public function testProcessWithDeprecation()
+    {
+        $container = new ContainerBuilder();
+        $fooDefinition = $container
+            ->register('deprecated_foo')
+            ->setPublic(false)
+        ;
+        $barDefinition = $container
+            ->register('bar')
+            ->setPublic(true)
+            ->setDeprecated(true)
+            ->setDecoratedService('deprecated_foo')
+        ;
+
+        $wasTriggered = false;
+        $that = $this;
+
+        set_error_handler(function ($errno, $errstr) use (&$wasTriggered, $that) {
+            $that->assertSame(E_USER_DEPRECATED, $errno);
+            $that->assertSame('The service deprecated_foo is deprecated. You should stop using it, as it will soon be removed.', $errstr);
+
+            $wasTriggered = true;
+        });
+
+        $this->process($container);
+
+        restore_error_handler();
+
+        $this->assertTrue($wasTriggered);
```

Comment:
Indeed, I put a breakpoint on the trigger and it seems that it is not run, so i'll check it out.